apps/arena/lib/arena/game_updater.ex

```diff
@@ -824,4 +934,23 @@ defmodule Arena.GameUpdater do
       end
     end)
   end
+
+  defp handle_pools(game_state) do
+    Enum.each(game_state.players, fn {player_id, player} ->
+      colliding_pools =
+        game_state.pools
+        |> Map.filter(fn {pool_id, _pool} -> pool_id in player.collides_with end)
+
+      Enum.each(game_state.pools, fn {pool_id, pool} ->
+        if Map.has_key?(colliding_pools, pool_id) do
+          send(
+            self(),
+            {:add_pool_effects, player_id, pool, pool.aditional_info.effects_to_apply}
+          )
+        else
+          send(self(), {:remove_pool_effects, player_id, pool_id})
```

Comment:
You do a `send/2` because you want to delay the addition/removal of the effects by one tick (on next one)?

Reply:
i found the message easier to do to remove the pool effects from a player, i found another walk around to avoid creating a lot of messages but im'n not entirely happy with the outcome, can you give me a helping hand when you've spare time pls?

Reply:
It's fine either way, was only wondering if it was a "product" decision or an implementation detail